sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,238 @@
+        /// or groups.
+        /// </summary>
+        public bool ExtendedInfoInAcl { get; set; }
+
+        /// <summary>
+        /// Internal empty constructor.
+        /// </summary>
+        public PathPermissions() { }
+
+        /// <summary>
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit,
+            bool extendedInfoInAcl)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octal">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octal)
+        {
+            if (octal == null)
+            {
+                throw Errors.ArgumentNull(nameof(octal));
+            }
+
+            if (octal.Length != 4)
+            {
+                throw Errors.InvalidArgument(nameof(octal));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (octal[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else
+            {
+                pathPermissions.StickyBit = true;
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[1]);
+            pathPermissions.Group = RolePermissionsExtensions.ParseOctal(octal[2]);
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="str">String to parse</param>
+        /// <returns><see cref="PathPermissions"/></returns>
+        public static PathPermissions ParseSymbolic(string str)
+        {
+            if (str == null)
+            {
+                throw Errors.ArgumentNull(nameof(str));
+            }
+
+            if (str.Length != 9 && str.Length != 10)
+            {
+                throw Errors.InvalidArgument(nameof(str));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            // Set sticky bit
+            if (str.ToLower(CultureInfo.InvariantCulture)[str.Length - 1] == 't')
```

Comment:
this checks the last character and not index 8(?)

Reply:
fixed